### PR TITLE
[PRMT-4817] NACKs remove conversationActivityTimestamp

### PR DESCRIPTION
**Pull request by AndyFlintNHS**

### CODE REVIEW

**Review by martin-nhs — APPROVED** [review]

**Review by chrisbloe — APPROVED** [review]
Happy with these changes... a few fixes in here :).

File: terraform/ecs-task.tf [hunk 1]
```diff
@@ -39,10 +39,9 @@ locals {
     { name = "SPLUNK_UPLOADER_TOPIC_ARN", value = aws_sns_topic.splunk_uploader.arn },
     { name = "TIMEOUT_DURATION_IN_SECONDS", value = var.timeout_in_seconds},
     { name = "TIMEOUT_SCHEDULER_FIXED_RATE_IN_SECONDS", value = var.timeout_scheduler_fixed_rate_in_seconds}
```
Comment (martin-nhs) [inline terraform/ecs-task.tf:41]: Missing a dreaded comma here! 😂

File: src/main/java/uk/nhs/prm/repo/ehrtransferservice/database/TransferService.java [hunk 2]
```diff
@@ -77,6 +77,10 @@ public void updateConversationTransferStatus(UUID inboundConversationId, Convers
 
     public void updateConversationTransferStatusWithFailure(UUID inboundConversationId, String failureCode) {
         transferRepository.updateConversationStatusWithFailure(inboundConversationId, failureCode);
+
+        // remove the in-memory activity to end the conversation
+        activityService.removeConversationActivityTimestamp(inboundConversationId);
```
Comment (martin-nhs) [inline src/main/java/uk/nhs/prm/repo/ehrtransferservice/database/TransferService.java:82]: Might be out of scope for this ticket, though myself and @MohammadIqbalAD-NHS (paired-review) agree that perhaps removing the `Timestamp` could enhance readability.

```suggestion
        activityService.removeConversationActivity(inboundConversationId);
```

File: src/main/java/uk/nhs/prm/repo/ehrtransferservice/handlers/LargeMessageFragmentHandler.java [hunk 3]
```diff
@@ -22,17 +22,17 @@ public void handleMessage(ParsedMessage fragmentMessage) throws Exception {
 
         if (storeMessageResult.isEhrComplete()) {
             log.info("Successfully stored all fragments for Inbound Conversation ID {}",
-                fragmentMessage.getConversationId());
+                    fragmentMessage.getConversationId().toString().toUpperCase());
```
[the remark this answers is not in the document]
Comment (AndyFlintNHS) [reply src/main/java/uk/nhs/prm/repo/ehrtransferservice/handlers/LargeMessageFragmentHandler.java:25]: Funnily enough, this is the correct formatting if you CTRL+ALT+L it! It was incorrect previously.

File: src/main/java/uk/nhs/prm/repo/ehrtransferservice/handlers/LargeMessageFragmentHandler.java [hunk 4]
```diff
@@ -22,17 +22,17 @@ public void handleMessage(ParsedMessage fragmentMessage) throws Exception {
...
 
     private StoreMessageResult storeFragmentMessage(ParsedMessage fragmentMessage) throws Exception {
         LargeEhrFragmentMessage largeEhrFragment = new LargeEhrFragmentMessage(fragmentMessage);
         StoreMessageResult storeMessageResult = ehrRepoService.storeMessage(largeEhrFragment);
-        log.info("Successfully stored fragment with Inbound Message ID {} for Outbound Conversation ID {}",
-            fragmentMessage.getMessageId(),
-            fragmentMessage.getConversationId()
+        log.info("Successfully stored fragment with Inbound Message ID {} for Inbound Conversation ID {}",
+                fragmentMessage.getMessageId().toString().toUpperCase(),
+                fragmentMessage.getConversationId().toString().toUpperCase()
```
[the remark this answers is not in the document]
Comment (AndyFlintNHS) [reply src/main/java/uk/nhs/prm/repo/ehrtransferservice/handlers/LargeMessageFragmentHandler.java:35]: See above

File: src/main/java/uk/nhs/prm/repo/ehrtransferservice/database/TransferService.java [hunk 5]
```diff
@@ -77,6 +77,10 @@ public void updateConversationTransferStatus(UUID inboundConversationId, Convers
 
     public void updateConversationTransferStatusWithFailure(UUID inboundConversationId, String failureCode) {
         transferRepository.updateConversationStatusWithFailure(inboundConversationId, failureCode);
+
+        // remove the in-memory activity to end the conversation
```
Comment (martin-nhs) [inline src/main/java/uk/nhs/prm/repo/ehrtransferservice/database/TransferService.java:81]: If `Timestamp` is removed from the below method call, this comment is unnessesary due to it being readable as-is:

- updateConversationTransferStatusWithFailure
- activityService.removeConversationActivity(inboundConversationId)

File: terraform/ecs-task.tf (hunk 1 above)
Comment (AndyFlintNHS) [reply terraform/ecs-task.tf:41]: EAGLE eyed! Very well spotted.

File: src/main/java/uk/nhs/prm/repo/ehrtransferservice/handlers/LargeMessageFragmentHandler.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (martin-nhs) [reply src/main/java/uk/nhs/prm/repo/ehrtransferservice/handlers/LargeMessageFragmentHandler.java:25]: All good :) Happy to ignore the other ones then @AndyFlintNHS!

File: src/main/java/uk/nhs/prm/repo/ehrtransferservice/services/ConversationActivityService.java [hunk 6]
```diff
@@ -9,6 +9,10 @@
 import java.util.UUID;
 import java.util.concurrent.ConcurrentHashMap;
 
+/**
+ * Capture a timestamp of the most recent activity for a given conversation ID. If the conversation has not been
+ * active for a set duration of time, the conversation status will be set to timed out. Upon the
```
Comment (MohammadIqbalAD-NHS) [inline src/main/java/uk/nhs/prm/repo/ehrtransferservice/services/ConversationActivityService.java:14]: This sentence hasn't been completed!

**Pull request merged**